docker_images/sklearn/app/pipelines/tabular_classification.py

```diff
@@ -1,18 +1,37 @@
+import json
+from pathlib import Path
 from typing import Dict, List, Union
 
 import joblib
-from app.pipelines import Pipeline
-from huggingface_hub import cached_download, hf_hub_url
+from huggingface_hub import snapshot_download
+from pipelines import Pipeline
 
 
 DEFAULT_FILENAME = "sklearn_model.joblib"
 
 
 class TabularDataPipeline(Pipeline):
     def __init__(self, model_id: str):
-        self.model = joblib.load(
-            open(cached_download(hf_hub_url(model_id, DEFAULT_FILENAME)), "rb")
-        )
+        print("a")
+        cached_folder = snapshot_download(repo_id=model_id)
+        print(cached_folder)
+        try:
+            with open(Path(cached_folder) / "config.json") as f:
+                # this is the default path for configuration of a scikit-learn
+                # project. If the project is created using `skops`, it should have
+                # this file.
+                config = json.load(f)
+            print(config)
+            model_file = config["sklearn"]["model"]["file"]
```

Comment:
It's not adding complexity really. We're just not forcing people with a specific name, and there doesn't seem to be anything really depending on the hard coded name. I'm not sure why the model should ideally always have the same name. What are we trying to infer from the name that makes it significant?

Reply:
>  the model should ideally always have the same name 

for transformers for instance it was IMO important for adoption to have a principle of least surprise/consistency/Developer experience.

Is there any advantage to be able to pick an arbitrary filename?

Reply:
To me, developer experience is why I want it to be flexible. scikit-learn is old and people have very different behaviors and habits when it comes to saving their models, and I don't see why we should enforce one over another.

Also, some people might save it with joblib and some with pickle, they're not even the same thing necessarily.

Reply:
I agree with @adrinjalali here, unlike transformers, there's not much consistency across the developer behavior of scikit-learn since everyone learns it from different sources. Some people call their models model, some call it clf, some call it pipeline, they also serialize differently.

Reply:
@merveenoyan you could say the same for TF in which some users use h5 and others SavedModel, but we decided to choose consistency :D 

The official docs suggest joblib or pickle, so I think supporting both formats is ok, but I would at least try to be consistent with the model filename. I don't see an advantage of letting any filename. Also the way they name the variable is totally ok and up to the users (users use `pipe`, `model`, `generator`, etc with `transformers` stuff), it's more about how the model file is named.

Reply:
Imagine this PR is coming from a developer of a package from the community, and they want to give their users this flexibility. They're implementing it, and it's not adding much complexity on our side, I don't see why we'd try to enforce HF ideas on them.

Reply:
@osanseviero Keras models are not serialized through .h5 for a while because google sort of bought it and provided better option to have SavedModel to leverage TFX. they literally only have two options and one of them is almost deprecated 😅 but people still want to use .h5 in Hub (though we do not support it) given we've seen PRs or issues on `huggingface_hub` repository and discord. If it doesn't add complexity on our side, I'm not in favor of forcing developer behavior in a monocultural way. We can call it X in our docs or encourage the use of X but not enforce it.

Reply:
Up to you :) you know better the ecosystem but please make sure to consider the trade-offs.

> I don't see why we'd try to enforce HF ideas on them.

Consistency has played a big role in adoption in the past. With the proposal, we might  see **many** beginners confused why their model does not work/load properly and the reason being that they don't have a config file with a sklearn key with a model key with a file key, which adds complexity and potential sources of errors for users. If we're confident `skops` will handle all of this very nicely then it might not be such a problem though :)

Reply:
I very much agree with all those arguments. I'm a firm believer of "simple things should be easy, complicated things should be possible". That's why I haven't removed the default name and I think a repo with the default name should still work. Although we still need to handle dependencies. But I completely agree with what you're trying to achieve.